core/include/kernel/dt.h

```diff
@@ -52,8 +52,13 @@ struct dt_device_match {
 	const char *compatible;
 };
 
+enum dt_driver_type {
+	DT_DRIVER_UART,
```

Comment:
Could it be useful to add a DT_DRIVER_NOTYPE = 0 that will allow to define a dt_driver_type without a type defined?
It could be useful to loop around a dt_driver list to probe them and avoid a mistake that the first define type is UART.

Reply:
I will add that yes.